internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -137,6 +142,11 @@ func (u *Upgrader) Upgrade(ctx context.Context, a Action, reexecNow bool) (_ ree
 	sourceURI := u.sourceURI(a.SourceURI())
 	archivePath, err := u.downloadArtifact(ctx, a.Version(), sourceURI)
 	if err != nil {
+		// Run the same preUpgradeCleanup task to get rid of any newly downloaded files
+		// This may have an issue if users are upgrading to the same version number.
+		if dErr := preUpgradeCleanup(a.Version()); dErr != nil {
+			u.log.Errorf("Unable to remove file after verification failure: %v", dErr)
+		}
 		return nil, err
```

Comment:
[Blocker]
 - Is it correct, the test is for `dErr`, but `err` is returned?
 - How would be possible to update to the same version? Anyway, if that is the case, the clean up is ineffective. Wouldn't be better just to remove all files on the folder? As I suggested above, we could have just the `cleanAllDownloads` and get rid of everything on every upgrade.

Reply:
yes, we don't want to fail the upgrade because an artifact can't be deleted
the artifacts in downloads are used if a new binary is added as an input